### PR TITLE
[#298] fix(library)!: allow arbitrarily complex custom arguments

### PR DESCRIPTION
**Pull request by krzema12**
A change in API was made so that it's more intuitive to create deeply
nested structures. It was possible thanks to using a classic YAML
emitter (snakeyaml) that works on reflection (contrary to
kotlinx.serialization).

### CODE REVIEW

File: library/src/main/kotlin/it/krzeminski/githubactions/domain/Job.kt [hunk 1]
```diff
@@ -15,7 +15,7 @@ data class Job(
     val strategyMatrix: Map<String, List<String>>? = null,
     val timeoutMinutes: Int? = null,
     val concurrency: Concurrency? = null,
-    override val _customArguments: Map<String, CustomValue> = mapOf(),
+    override val _customArguments: Map<String, @Contextual Any> = mapOf(),
```
Comment (jmfayard) [inline library/src/main/kotlin/it/krzeminski/githubactions/domain/Job.kt:18]: What does this do?

**Review by jmfayard — COMMENTED** [review]
Are there no other breaking changes for the script-generator than those `ListCustomValue.asClassName() -> Any::Class.asClassName()` ?
That sounds too good to be true :)

**Comment by krzema12** [comment]
> Are there no other breaking changes for the script-generator than those `ListCustomValue.asClassName() -> Any::Class.asClassName()` ? That sounds too good to be true :)

The unit tests show no regression 😄

File: library/src/main/kotlin/it/krzeminski/githubactions/domain/Job.kt (hunk 1 above)
Comment (krzema12) [reply library/src/main/kotlin/it/krzeminski/githubactions/domain/Job.kt:18]: Try to remove it, you will see :)

In general the kotlinx.serialization annotations in the lib are needed only for the script generator.

**Comment by krzema12** [comment]
@jmfayard I'd like to include this fix in the release tomorrow so let me merge it, and in case of any issues with the script generator we can always fix it later. Looking at the unit tests, though, it seems to work fine 😃

**Comment by jmfayard** [comment]
Ok!

**Pull request merged**